server/util/log.js

```diff
@@ -13,15 +13,19 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
 
-var recentErrors = [];
-var ERROR_BUFFER_SIZE = 100;
+import * as time from './time.js';
+import EventEmitter from 'events';
 
-var pushError = function(record) {
+export const recentErrors = [];
+const ERROR_BUFFER_SIZE = 100;
+
+function pushError(record) {
   recentErrors.push(record);
-  if (recentErrors.length > ERROR_BUFFER_SIZE) {
+  emitter.emit('error', record);
+  while (recentErrors.length > ERROR_BUFFER_SIZE) {
     recentErrors.shift();
```

Comment:
optional: emit a log that we "dropped N errors" when you do this?

Reply:
It will always be one with the current implementation, so not worth it.